joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/passes/ObjectPropertyCallLinker.scala

```diff
@@ -0,0 +1,42 @@
+package io.joern.jssrc2cpg.passes
+
+import io.shiftleft.codepropertygraph.generated.nodes.{Call, MethodRef}
+import io.shiftleft.codepropertygraph.generated.{Cpg, PropertyNames}
+import io.shiftleft.passes.CpgPass
+import overflowdb.BatchedUpdate
+import io.shiftleft.semanticcpg.language.*
+
+/** Perform a simple analysis to find a common pattern in JavaScript where objects are dynamically assigned function
+  * pointers. To keep this precise, this will only match objects defined within the scope of the same file.
+  *
+  * This relies on JavaScriptTypeHintCallLinker.
+  */
+class ObjectPropertyCallLinker(cpg: Cpg) extends CpgPass(cpg) {
+
+  override def run(builder: BatchedUpdate.DiffGraphBuilder): Unit = {
+    val propertyCallRegex = "^(?:\\{.*\\}|.*<returnValue>):<member>\\((.*)\\):.*$".r
+    val objectCalls       = cpg.call.methodFullName("^(?:\\{.*\\}|.*<returnValue>):<member>\\(.*\\):.*$").l
```

Comment:
Isnt that `propertyCallRegex`?

Reply:
`\\(.*\\)` versus `\\((.*)\\)`

Reply:
Might be useful to combine this into a regex generator function to reduce duplication